DSCResources/MSFT_xWebSiteDefaults/MSFT_xWebSiteDefaults.psm1

```diff
@@ -0,0 +1,176 @@
+        [string]$DefaultApplicationPool,
+        [ValidateSet("true","false")]
+        [string]$AllowSubDirConfig
+    )
+
+        CheckIISPoshModule
+
+        SetValue "siteDefaults/logFile" "logFormat" $LogFormat
+        SetValue "siteDefaults/logFile" "directory" $LogDirectory
+        SetValue "siteDefaults/traceFailedRequestsLogging" "directory" $TraceLogDirectory
+        SetValue "applicationDefaults" "applicationPool" $DefaultApplicationPool
+        SetValue "virtualDirectoryDefaults" "allowSubDirConfig" $AllowSubDirConfig
+}
+
+######################################################################################
+# The Test-TargetResource cmdlet.
+# This will test whether all given values are already set in the current configuration
+######################################################################################
+function Test-TargetResource
+{
+    [OutputType([System.Boolean])]
+	param
+	(	
+        [ValidateSet("Machine")]
+        [parameter(Mandatory = $true)]
+        [string]$ApplyTo,
+        [ValidateSet("W3C","IIS","NCSA","Custom")]
+        [string]$LogFormat,
+        [string]$LogDirectory,
+        [string]$TraceLogDirectory,
+        [string]$DefaultApplicationPool,
+        [ValidateSet("true","false")]
+        [string]$AllowSubDirConfig
+	)
+
+    [bool]$DesiredConfigurationMatch = $true;
+
+    CheckIISPoshModule
+
+    $DesiredConfigurationMatch = CheckValue "virtualDirectoryDefaults" "allowSubDirConfig" $AllowSubDirConfig
+    if (!($DesiredConfigurationMatch)) { return $false }
+
+    $DesiredConfigurationMatch = CheckValue "siteDefaults/logFile" "logFormat" $LogFormat
+    if (!($DesiredConfigurationMatch)) { return $false }
+
+    $DesiredConfigurationMatch = CheckValue "siteDefaults/logFile" "directory" $LogDirectory
+    if (!($DesiredConfigurationMatch)) { return $false }
+
+    $DesiredConfigurationMatch = CheckValue "siteDefaults/traceFailedRequestsLogging" "directory" $TraceLogDirectory
+    if (!($DesiredConfigurationMatch)) { return $false }
+
+    $DesiredConfigurationMatch = CheckValue "applicationDefaults" "applicationPool" $DefaultApplicationPool
+    if (!($DesiredConfigurationMatch)) { return $false }
+    
+	return $DesiredConfigurationMatch
+}
+
+Function CheckValue([string]$path,[string]$name,[string]$newValue)
+{
+    if (!$newValue)
+    {
+        # if no new value was specified, we assume this value is okay.        
+        return $true
+    }
+
+
+    [bool]$DesiredConfigurationMatch = $true;
+
+    $existingValue = GetValue $path $name
+    if ($existingValue -ne $newValue)
+    {
+        $DesiredConfigurationMatch = $false
+    }
+    else
+    {
+        $relPath = $path + "/" + $name
+        Write-Verbose($LocalizedData.ValueOk -f $relPath,$newValue);
+    }
+    
+    return $DesiredConfigurationMatch
+}
+
+# some internal helper function to do the actual work:
+
+Function SetValue([string]$path,[string]$name,[string]$newValue)
+{
+    if ($newValue)
```

Comment:
I'd probably just make $newValue mandatory and notnullorempty and remove this check. Given your later check of if  ($existingValue) I think your covered.